vars/kaGit.groovy

```diff
@@ -176,3 +176,68 @@ def safeMergeFromMaster(dir, commitToMergeInto, submodules=[]) {
             dir, commitToMergeInto] + _submodulesArg(submodules));
    }
 }
+
+
+// Merges multiple branches together into a single commit.
+// 
+// Arguments:
+// - gitRevisions: string containing one or more branche names separated by "+"
+// - tagName: string to tag the resulting commit with
+//
+// Notes:
+// - Used by merge-granches.groovy and deploy-znd.groovy.
+def mergeBranches(gitRevisions, tagName) {
+   // We don't use kaGit for many of the ops here, and use lower-level ops
+   // where we do. We can afford this because we don't need to update
+   // submodules at each step, and we don't need a fully clean checkout.  All
+   // we need is enough to merge.  This saves us a *lot* of time traversing all
+   // the submodules on each branch, and being careful to clean at each step.
+   def allBranches = gitRevisions.split(/\+/);
+   kaGit.quickClone("git@github.com:Khan/webapp", "webapp",
+                    allBranches[0].trim());
+   dir('webapp') {
+      // We need to reset before fetching, because if a previous incomplete
+      // merge left .gitmodules in a weird state, git will fail to read its
+      // config, and even the fetch can fail.  This also avoids certain
+      // post-merge-conflict states where git checkout -f doesn't reset as much
+      // as you might think.
+      exec(["git", "reset", "--hard"]);
+   }
+   kaGit.quickFetch("webapp");
```

Comment:
I think this needs to be just `quickFetch` now?  I'm not 100% sure.

And likewise below.